packages/wonder-blocks-form/src/components/text-field.tsx

```diff
@@ -248,6 +269,12 @@ class TextField extends React.Component<PropsWithForwardRef, State> {
             ...otherProps
         } = this.props;
 
+        if ((min || max || step) && type !== "number") {
+            throw new Error(
+                'The props `min`, `max`, and `step` can only be used with `type="number"`.',
+            );
+        }
```

Comment:
**thought:** this should probably be checked at build time instead of runtime. Probably using a TS discriminated union type would be better for this case.

You can see an example of this here: https://www.typescriptlang.org/play/?&q=387#code/C4TwDgpgBAwg9gWwXAdgBQE5zAZygXigG8AoKcqAegCpqyKpqoAVAC2gFcUBLARw+jcAJhBTBuAM24QMUCXFnB2UbijAdgAOnoVqlHSqEAuKDmAZVAcwDcBmnQaMWy1euBQAbgEMANgO2OegbefhAmZhYoNna0Bk5s0CheCNDyii5qGlAAxqjmcD6aztx4JaYcAEYI3MDAEEJQvqiWUADuNaxxTEqpCghQQl7AXgEMQQxJKQD84eZWtgC+tiSgkFAA8j0YAJKZwJjYeITwSKgHuFAAZMQGqxAz5ABEdQAewI9QAD5Qj2BeODhWgohB9vo8IAgvNwfKCfnUYYtlncoAA5DgpCzZXZuc5HWCIZDoLAXa6kBh3B6PFDoioyR62Bj2LrOaDVHgIdFQakY7jZTy+ARyBRQHoqPajXT6Bhsh7c2kYBkUJmBFlQSEvbgc-rcmS8-mhIXpQTirpSijq2U0mSK8jKsaqnWY-WC4BwMXZDAQ0TuYUiD1esRQCogCXkJgAJQg-G4nrwotcWVdQdZHB84jAPggeDgEhFrDKIX8poMZggYEtCHliJIK3A0FxBA2W2xGgb3zRPKxe1xy1yKDMIogbxbwBMDcIZIowhMjxqOAAtF55693gAaYICsI-HCICDLodr251mcrx4kJY1vsD1QhYTMA8jsfEvGT8jTn5zxf7t6PdcMQtbo8O4pN+h7ksecIHr+BhsiYACM64XiQV7uNyj5QOONwMO+s7AAuS6nn+FAATOwF7oRR6QDOcp0kR5CwVACHnkAA